docs/proposals/hpa/federated-hpa.md

```diff
@@ -0,0 +1,135 @@
+
+#### Story 2
+For an application developer, I create a HPA CR for the application running on Karmada with FederatedHPA enabled.
+```
+target cpu util 30%
+min replica 3
+max replica 100
+```
+Suddenly, one of the member clusters which my application running on stops to work and can't scale up new pods. Unfortunately, a request burst is comming into the application. The CPU util of pods becomes higher than 30%. It will need 100 Pods totally to take the request burst. Thanks to the good design of the Karmada FederatedHPA, the FederatedHPAController detects failure and modify the HPA CR in another healthy cluster to increase the `max` value. Finally, new pods are scaled on another healthy member cluster successfully. 
+
+#### Story 3
+As an administrator of the Karmada&Kubernetes platform, I receive an alert that the Karmada control plane stops to work and any requests to the Karmada control plane are failed. There are many applications running on the platform heavilly depend on the HPA to handle the unpredictable burst of requests. The chance of RCA occurred becomes really high if the system can't tolerate the failure of federation control plane.
+Thanks to the design of Karmada FederatedHPA, the single cluster version of HPA is reused and still works as expected in the member cluster without the help of Karmada control plane or any other federated controller.
+
+### Notes/Constraints/Caveats (Optional)
+1. The workloads/pods in different member clusters selected by the same HPA CR share the load of the application equally. For example, 10 pods of the application are spread into two member clusters with distribution `cluster1: 3 pods, cluster2: 7 pods`, so the 3 pods in cluster1 take 3/10 of total requests and 7 pods in cluster2 take 7/10 of total requests. Scenarios don't meet the restriction are not considered in this proposal.
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate? 
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+## Design Details
+### Architecture
+![Karmada FederatedHPA Architecture](statics/karmadafederatedhpa-arch.drawio.png)
+
+1. The Kubernetes related HPA components are still used in the member cluster and can work standalone.
+1. The FederatedHPAController is responsible for purposes
+   1. Distribute 'some' fields of the HPA CRs between member clusters based on the `Propagation`.
+   2. Redistribute 'some' fields of the HPA CRs after disaster of member clusters detected or `Propagation` is changed.
```

Comment:
How to detect the member cluster‘s fault in our design? Is it real-time?

Reply:
It depends on `Karmada`, which has already solved your question.

Reply:
Maybe we can add some details in case someone not familiar with this.

> @kerthcet Thanks for your comments, I hope I have answered your questions.

Sorry for the delayed reply, Looking forward to your detailed design.